### DISCUSSION TITLE
输入类控件支持面性、下划线等变体

### DISCUSSION THREAD

**Discussion by arvinxx** [ant-design/ant-design#43464] [RFCs]
## 背景

业务中会存在需要面性输入框、下划线选择器等诉求，目前的 antd 组件只支持线性和无边框两种样式变体。本 RFC 期望为输入类控件添加面性、下划线这两个变体，以适应不同场景下的输入诉求。

## 相关参考

maintine中面性叫 filled 

![image](https://github.com/ant-design/ant-design/assets/28616219/06711dee-8950-482a-a516-1ade332a397a)

MUI 没有面型

acro 和 semi ，默认都是面型，不支持切换

![image](https://github.com/ant-design/ant-design/assets/28616219/2404a4ce-843c-4099-a50c-e27137b1d9f0)

nextjs 的支持的很丰富，默认、面性、下划线都有

![image](https://github.com/ant-design/ant-design/assets/28616219/200317cc-b17d-4fbb-8d1f-47b195d8e960)

## 建议 api

type variant = "default" | "filled" | "underline" | "borderless"

**Comment by MadCcc** [comment]
面性的就没有 focus 样式了么

**Reply by arvinxx** [reply]
focus 样式复用之前的应该没问题？

**Reply by MadCcc** [reply]
感觉灰底会不太好看诶

**Reply by arvinxx** [reply]
你说的 focus 是键盘的 focus 吗

**Reply by latte03** [reply]
面性 focus 也可以通过border，acro focus的时候是白底。

**Comment by MadCcc** [comment]
```tsx
type InputProps = {
  type?: 'outlined' | 'filled' | 'underlined' | 'borderless',
  /** @deprecated */
  bordered?: boolean
}
```

要把 borderless 收进去的话 `bordered` 就要废弃了。现在 `bordered` 这个 API 用的还挺多的

**Reply by arvinxx** [reply]
input 有原生 type 吗？会不会冲突

**Comment by vitalijalbu** [comment]
Would be fine having them all... 👀 ANTD need to grow to be the 1st used library 😁

**Comment by zombieJ** [comment]
Conclusion: use `variant` prop name.
